src/testbase.rs

```diff
@@ -193,7 +206,55 @@ fn traverse_some_built_bh<BH: BoundingHierarchy<f32, 3>>(all_shapes: &[UnitBox],
         expected_shapes.insert(4);
         expected_shapes.insert(5);
         expected_shapes.insert(6);
-        traverse_and_verify(origin, direction, all_shapes, &bh, &expected_shapes);
+        traverse_and_verify(
+            &TRay3::new(origin, direction),
+            all_shapes,
+            &bh,
+            &expected_shapes,
+        );
+    }
+
+    {
+        // Define a point at the origin.
+        let point = TPoint3::new(0.0, 0.0, 0.0);
+
+        // It should be contained by the middle box.
+        let mut expected_shapes = HashSet::new();
+        expected_shapes.insert(0);
+        traverse_and_verify(&point, all_shapes, &bh, &expected_shapes);
+    }
+
+    {
+        // Define a point far away.
+        let point = TPoint3::new(0.0, 1000.0, 0.0);
+
+        // It shouldn't be contained by any boxes.
+        let expected_shapes = HashSet::new();
+        traverse_and_verify(&point, all_shapes, &bh, &expected_shapes);
+    }
+
+    {
+        // Define an AABB intersecting with some boxes.
+        let aabb = Aabb::with_bounds(TPoint3::new(5.1, -1.0, -1.0), TPoint3::new(9.9, 1.0, 1.0));
+
+        // It shouldn't be contained by any boxes.
+        let mut expected_shapes = HashSet::new();
+        for x in 5..=10 {
+            expected_shapes.insert(x);
+        }
+        traverse_and_verify(&aabb, all_shapes, &bh, &expected_shapes);
+    }
+
+    {
+        // Define a sphere intersecting with some boxes.
+        let aabb = Sphere::new(TPoint3::new(5.0, -1.0, -1.0), 1.4);
+
+        // It shouldn't be contained by any boxes.
+        let mut expected_shapes = HashSet::new();
+        for x in 4..=6 {
+            expected_shapes.insert(x);
+        }
+        traverse_and_verify(&aabb, all_shapes, &bh, &expected_shapes);
```

Comment:
Man, this old test setup I had here is so crusty. Nowadays I'd solve this using integration testing and rstest. It's fine for you to continue in this fashion for now but we should really look into modernizing this soonish.